README.md
<!-- README.md is generated from README.Rmd. Please edit that file -->
[![Build Status](https://api.travis-ci.org/mskyttner/swedishbirdtrends.png)](https://travis-ci.org/mskyttner/swedishbirdtrends)

This is a package - `swedishbirdtrends` - that distributes Swedish bird population trend data (currently from 2016-02-16) from the Swedish Bird Inventory Facility <http://www.fageltaxering.lu.se> along with some interactive web user interfaces presenting this data.

The package does not provide real-time, online bird monitoring data and therefore it can be used offline.

Install
-------

The development version from Github

``` r
install.packages("devtools")
devtools::install_github("ropensci/plotly")
devtools::install_github("mskyttner/swedishbirdtrends")
```

Then load the package

``` r
suppressWarnings(suppressPackageStartupMessages(library(swedishbirdtrends)))
```

Use Swedish bird population trend data
--------------------------------------

Inspect the bundled data:

``` r

data("birdtotals")
data("birdtrends")

birdtotals
#> Source: local data frame [477 x 8]
#> 
#>      Art   Arthela        Ind YPctChg Significance Series Suffix
#>    (chr)     (chr)      (dbl)   (dbl)        (chr)  (chr)  (chr)
#> 1    008 Storskarv   923.0513    3.29          *** Vinter     NA
#> 2    009     Häger   133.7179    3.63          *** Vinter     NA
#> 3    012   Gräsand 10231.1538    2.60          *** Vinter     NA
#> 4    016   Bläsand   414.0000   27.19              Vinter     NA
#> 5    020      Vigg  9798.0256    2.93          *** Vinter     NA
#> 6    021   Brunand   224.8974    3.81            * Vinter     NA
#> 7    022     Knipa  2001.8462    1.06          *** Vinter     NA
#> 8    023   Alfågel  1657.7692   -1.66              Vinter     NA
#> 9    026     Ejder   749.4615    0.96              Vinter     NA
#> 10   027 Småskrake   191.1026    0.73              Vinter     NA
#> ..   ...       ...        ...     ...          ...    ...    ...
#> Variables not shown: Vernacular (chr)
birdtrends
#> Source: local data frame [12,598 x 8]
#> 
#>      Art   Arthela  Year Measure         ts Series Suffix Vernacular
#>    (chr)     (chr) (chr)   (dbl)     (time)  (chr)  (chr)      (chr)
#> 1    008 Storskarv  1975  0.3378 1975-01-01 Vinter     NA  Storskarv
#> 2    009     Häger  1975  0.9471 1975-01-01 Vinter     NA      Häger
#> 3    012   Gräsand  1975  0.4342 1975-01-01 Vinter     NA    Gräsand
#> 4    016   Bläsand  1975      NA 1975-01-01 Vinter     NA    Bläsand
#> 5    020      Vigg  1975  0.5928 1975-01-01 Vinter     NA       Vigg
#> 6    021   Brunand  1975  0.1161 1975-01-01 Vinter     NA    Brunand
#> 7    022     Knipa  1975  0.3211 1975-01-01 Vinter     NA      Knipa
#> 8    023   Alfågel  1975  0.3505 1975-01-01 Vinter     NA    Alfågel
#> 9    026     Ejder  1975  1.8316 1975-01-01 Vinter     NA      Ejder
#> 10   027 Småskrake  1975  0.5940 1975-01-01 Vinter     NA  Småskrake
#> ..   ...       ...   ...     ...        ...    ...    ...        ...
```

Explore Swedish Bird Population trends interactively
----------------------------------------------------

``` r
runShinyApp("birdtrends")
```

Compare Swedish Bird species longterm trends
--------------------------------------------

``` r
runShinyApp("birdtotals")
```

Meta
----

-   Please [report any issues or bugs](https://github.com/mskyttner/swedishbirdtrends/issues).
-   License: AGPL


NEWS.md
# swedishbirdtrends


For more fine-grained list of changes or to report a bug, consult 

* [The issues log](https://github.com/mskyttner/swedishbirdtrends/issues)
* [The commit log](https://github.com/mskyttner/swedishbirdtrends/master)

## v0.0.0.9006

* Retrieve only images (not audio media) from The Naturalist
* New layout for Species trend report following layout feedback
* Add table of contents report

## v0.0.0.9005

* Add loess smoother to static trend line displays, documented in vignette
* Add scientific names to datasets and recompress to save space
* Removed non-birds from datasets
* Add birduuids mapping bird species to identifiers used on The Naturalist (naturforskaren.se)
* Add vignette sbt-species-vignette showing generation of a Tufte styled bird population trend report, the report template is parameterized on the vernacular name
* Add script to generate species reports for all bird species in the package. This can be used to distribute bird population trend reports in the form of a static Jekyll (Tufte-Jekyll) website

## v0.0.0.9004

* Updated README.md with installation instructions
* Config changes - trying to get Travis CI with GitHub Releases to work. 

## v0.0.0.9002

* Various configuration fixes related to Travis CI

## v0.0.0.9000

* First version with data bundled from Swedish Bird Survey and a Shiny app for exploring the data.
* Added a `NEWS.md` file to track changes to the package.

R/refdata.R
#' Bird population trends in Sweden
#'
#' A dataset containing bird population trend indices from Sweden
#' This dataset is from 2016-02-16
#' 
#' @format A data frame [12,598 x 6]
#' \describe{
#'   \item{Art}{Species Number}
#'   \item{Arthela}{Species Name}
#'   \item{Year}{Year}
#'   \item{Measure}{Measure for index}
#'   \item{ts}{Timestamp for the year}
#'   \item{Series}{Series - route used - Winter, Summer, Standard or Night}
#'   \item{Suffix}{Suffix - qualifier - such as (S) for South, (N) for North, ob for not determined etc}
#'   \item{Vernacular}{Vernacular or Popular name - the common name for the species, cleaned from any qualifier or suffix}
#'   \item{Scientific}{Scientific name, the latin binomen}
#' }
#' @source \url{http://www.fageltaxering.lu.se/sites/default/files/files/Data/populationsindex.xls}
"birdtrends"

#' Bird population totals in Sweden
#'
#' A dataset containing bird population trend indices from Sweden
#' This dataset is from 2016-02-16
#' 
#' @format A data frame [477 x 7]
#' \describe{
#'   \item{Art}{Art numeric code identifier}
#'   \item{Arthela}{The whole Swedish species name}
#'   \item{Ind}{Indicator}
#'   \item{YPctChg}{Yearly change in percent}
#'   \item{Significance}{Statistical significance where * is lower and *** is higher}
#'   \item{Series}{Which route was used for the inventory - Vinter, Sommar, Standard or Natt}
#'   \item{Suffix}{The suffix contains some special abbreviations used in Arthela}
#'   \item{Vernacular}{Vernacular or Popular name - the common name for the species, cleaned from any qualifier or suffix}
#'   \item{Scientific}{Scientific name, the latin binomen}
#' }
#' @source \url{http://www.fageltaxering.lu.se/sites/default/files/files/Data/populationsindex.xls}
"birdtotals"


#' Bird identifiers mapped to unique identifiers used at The Naturalist / Naturforskaren (naturforskaren.se)
#'
#' A dataset containing bird population trend indices from Sweden
#' This dataset is from 2016-02-16
#' 
#' @format A data frame [22,840 x 2]
#' \describe{
#'   \item{species_uuid}{unique identifier for a species used in The Naturalist}
#'   \item{vernacular}{vernacular name in lower case}
#'   \item{sciname}{Scientific name}
#' }
#' @source \url{http://naturforskaren.se}
"birduuids"
